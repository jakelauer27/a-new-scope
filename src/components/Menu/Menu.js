import React from 'react'

export default function Menu({ changePage }) {
  return (
    <div className='menu'>
      <h1 className='main-title'>a new scope</h1>
      <div className='favorites-path'>
        <div 
          onClick={() => changePage('favorites')}
          className='favorites-planet'
        >
        <h3 className='planet-label'>Favorites</h3>
        </div>
        <div className='vehicles-path'>
          <div
            onClick={() => changePage('vehicles')}
            className='vehicles-planet'
          >
          <h3 className='planet-label'>Vehicles</h3>
          </div>
          <div className='planets-path'>
            <div
              onClick={() => changePage('planets')}
              className='planets-planet'
            >
            <h3 className='planet-label'>Planets</h3>
            </div>
            <div className='people-path'>
              <div
                onClick={() => changePage('people')}
                className='people-planet'
              >
              <h3 className='planet-label'>People</h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className='alt-menu-container'>
        <div 
          onClick={() => changePage('favorites')}
          className='alt-menu alt-menu-favorites'
        >
          <h3>Favorites</h3>
        </div>
        <div
          onClick={() => changePage('vehicles')}
          className='alt-menu alt-menu-vehicles'
        >
          <h3>Vehicles</h3>
        </div>
        <div
            onClick={() => changePage('planets')}
            className='alt-menu alt-menu-planets'
          >
          <h3>Planets</h3>
        </div>
        <div
          onClick={() => changePage('people')}
          className='alt-menu alt-menu-people'
        >
          <h3>People</h3>
        </div>
      </div>
    </div>
  )
}