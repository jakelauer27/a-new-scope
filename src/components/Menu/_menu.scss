.menu {
  width: 650px;
  height: 655px;
  margin: 30px auto;
}

@mixin path($height, $width, $top, $left) {
  border: 5px solid rgba(0, 208, 255, 0.99);
  border-radius: 50%;
  box-shadow: 0 0 2px 1px #00D1FF,
              0 0 15px 2px #00D1FF,
              0 0 25px 1px #00D1FF,
        inset 0 0 2px 1px #00D1FF,
        inset 0 0 25px 1px #00D1FF,
        inset 0 0 15px 2px #00D1FF;
  height: $height;
  width: $width;
  position: relative;
  top: $top;
  left: $left;
  z-index: 3;
}

@mixin planet($top, $left) {
  border: 5px solid rgba(0, 208, 255, 0.99);
  background: rgba(0, 208, 255, 0.99);
  width: 35px;
  height: 35px;
  border-radius: 50%;
  box-shadow: 0 0 2px 1px #00D1FF,
              0 0 15px 2px #00D1FF,
              0 0 25px 1px #00D1FF,
        inset 0 0 2px 1px #00D1FF,
        inset 0 0 25px 1px #00D1FF,
        inset 0 0 15px 2px #00D1FF;
  position: absolute;
  top: $top;
  left: $left;
  transition: .5s ease;
  z-index: 4;
}

h3 {
  font-size: 1.25rem;
  color: black;
  position: relative;
  top: 2px;
  right: 101px;
  z-index: 5;
  width: 110px;
  text-align: center;
  padding: 5px;
  border-radius: 15px;
  transform: rotate(-20deg);
  background: rgba(0, 208, 255, 0.894);
  text-shadow: 0 0 3px rgba(0, 0, 0, 0.4);
}

.favorites-path {
  @include path(620px, 620px, 0, 10px);
}

.vehicles-path {
  @include path(500px, 500px, 67px, 43px);
}

.planets-path {
  @include path(350px, 350px, 80px, 90px);
}

.people-path {
  @include path(200px, 200px, 50px, 47px);
}

.favorites-planet {
  @include planet(287px, 289px);
  animation: slideIn 150s linear infinite;

  &:hover {
    filter: drop-shadow(0 0 0.75rem #00D1FF);
    transition: .5s ease;
    cursor: pointer;
  }
}

.vehicles-planet {
  @include planet(226px, 228px);
  animation: slideIn2 140s linear infinite;

  &:hover {
    filter: drop-shadow(0 0 0.75rem #00D1FF);
    transition: .5s ease;
    cursor: pointer;
  }
}

.planets-planet {
  @include planet(150px, 150px);
  animation: slideIn3 130s linear infinite;

  &:hover {
    filter: drop-shadow(0 0 0.75rem #00D1FF);
    transition: .5s ease;
    cursor: pointer;
  }
}

.people-planet {
  @include planet(77px, 78px);
  animation: slideIn4 120s linear infinite;

  &:hover {
    filter: drop-shadow(0 0 0.75rem #00D1FF);
    transition: .5s ease;
    cursor: pointer;
  }
}

@keyframes slideIn {
    from { 
      transform: rotate(0deg) translateX(312px) rotate(0deg);
    }
    to { 
      transform: rotate(360deg) translateX(312px) rotate(-360deg); 
    }
}

@keyframes slideIn2 {
    from { 
      transform: rotate(30deg) translateX(250px) rotate(-30deg);
    }
    to { 
      transform: rotate(390deg) translateX(250px) rotate(-390deg); 
    }
}

@keyframes slideIn3 {
    from { 
      transform: rotate(120deg) translateX(175px) rotate(-120deg);
    }
    to { 
      transform: rotate(480deg) translateX(175px) rotate(-480deg); 
    }
}

@keyframes slideIn4 {
    from { 
      transform: rotate(0deg) translateX(101px) rotate(0deg);
    }
    to { 
      transform: rotate(360deg) translateX(101px) rotate(-360deg); 
    }
}