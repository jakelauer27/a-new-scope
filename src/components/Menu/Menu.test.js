import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme'

describe('Menu', () => {
  const mockChange = jest.fn()
  const wrapper = shallow(<Menu changePage={mockChange}/>)

  it('should run change page when a planet is clicked', () => {
    wrapper.find('.vehicles-planet').simulate('click')
    expect(mockChange).toBeCalled()
    expect(mockChange).toBeCalledWith('vehicles')

    wrapper.find('.favorites-planet').simulate('click')
    expect(mockChange).toBeCalled()
    expect(mockChange).toBeCalledWith('favorites')

    wrapper.find('.people-planet').simulate('click')
    expect(mockChange).toBeCalled()
    expect(mockChange).toBeCalledWith('people')

    wrapper.find('.planets-planet').simulate('click')
    expect(mockChange).toBeCalled()
    expect(mockChange).toBeCalledWith('planets')
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})