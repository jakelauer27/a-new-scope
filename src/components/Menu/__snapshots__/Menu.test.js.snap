// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    changePage={
      [MockFunction] {
        "calls": Array [
          Array [
            "vehicles",
          ],
          Array [
            "favorites",
          ],
          Array [
            "people",
          ],
          Array [
            "planets",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title"
        >
          a new scope
        </h1>,
        <div
          className="favorites-path"
        >
          <div
            className="favorites-planet"
            onClick={[Function]}
          >
            <h3>
              Favorites
            </h3>
          </div>
          <div
            className="vehicles-path"
          >
            <div
              className="vehicles-planet"
              onClick={[Function]}
            >
              <h3>
                Vehicles
              </h3>
            </div>
            <div
              className="planets-path"
            >
              <div
                className="planets-planet"
                onClick={[Function]}
              >
                <h3>
                  Planets
                </h3>
              </div>
              <div
                className="people-path"
              >
                <div
                  className="people-planet"
                  onClick={[Function]}
                >
                  <h3>
                    People
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ],
      "className": "menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "a new scope",
          "className": "main-title",
        },
        "ref": null,
        "rendered": "a new scope",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="favorites-planet"
              onClick={[Function]}
            >
              <h3>
                Favorites
              </h3>
            </div>,
            <div
              className="vehicles-path"
            >
              <div
                className="vehicles-planet"
                onClick={[Function]}
              >
                <h3>
                  Vehicles
                </h3>
              </div>
              <div
                className="planets-path"
              >
                <div
                  className="planets-planet"
                  onClick={[Function]}
                >
                  <h3>
                    Planets
                  </h3>
                </div>
                <div
                  className="people-path"
                >
                  <div
                    className="people-planet"
                    onClick={[Function]}
                  >
                    <h3>
                      People
                    </h3>
                  </div>
                </div>
              </div>
            </div>,
          ],
          "className": "favorites-path",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                Favorites
              </h3>,
              "className": "favorites-planet",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="vehicles-planet"
                  onClick={[Function]}
                >
                  <h3>
                    Vehicles
                  </h3>
                </div>,
                <div
                  className="planets-path"
                >
                  <div
                    className="planets-planet"
                    onClick={[Function]}
                  >
                    <h3>
                      Planets
                    </h3>
                  </div>
                  <div
                    className="people-path"
                  >
                    <div
                      className="people-planet"
                      onClick={[Function]}
                    >
                      <h3>
                        People
                      </h3>
                    </div>
                  </div>
                </div>,
              ],
              "className": "vehicles-path",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Vehicles
                  </h3>,
                  "className": "vehicles-planet",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vehicles",
                  },
                  "ref": null,
                  "rendered": "Vehicles",
                  "type": "h3",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="planets-planet"
                      onClick={[Function]}
                    >
                      <h3>
                        Planets
                      </h3>
                    </div>,
                    <div
                      className="people-path"
                    >
                      <div
                        className="people-planet"
                        onClick={[Function]}
                      >
                        <h3>
                          People
                        </h3>
                      </div>
                    </div>,
                  ],
                  "className": "planets-path",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h3>
                        Planets
                      </h3>,
                      "className": "planets-planet",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Planets",
                      },
                      "ref": null,
                      "rendered": "Planets",
                      "type": "h3",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="people-planet"
                        onClick={[Function]}
                      >
                        <h3>
                          People
                        </h3>
                      </div>,
                      "className": "people-path",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h3>
                          People
                        </h3>,
                        "className": "people-planet",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "People",
                        },
                        "ref": null,
                        "rendered": "People",
                        "type": "h3",
                      },
                      "type": "div",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title"
          >
            a new scope
          </h1>,
          <div
            className="favorites-path"
          >
            <div
              className="favorites-planet"
              onClick={[Function]}
            >
              <h3>
                Favorites
              </h3>
            </div>
            <div
              className="vehicles-path"
            >
              <div
                className="vehicles-planet"
                onClick={[Function]}
              >
                <h3>
                  Vehicles
                </h3>
              </div>
              <div
                className="planets-path"
              >
                <div
                  className="planets-planet"
                  onClick={[Function]}
                >
                  <h3>
                    Planets
                  </h3>
                </div>
                <div
                  className="people-path"
                >
                  <div
                    className="people-planet"
                    onClick={[Function]}
                  >
                    <h3>
                      People
                    </h3>
                  </div>
                </div>
              </div>
            </div>
          </div>,
        ],
        "className": "menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "a new scope",
            "className": "main-title",
          },
          "ref": null,
          "rendered": "a new scope",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="favorites-planet"
                onClick={[Function]}
              >
                <h3>
                  Favorites
                </h3>
              </div>,
              <div
                className="vehicles-path"
              >
                <div
                  className="vehicles-planet"
                  onClick={[Function]}
                >
                  <h3>
                    Vehicles
                  </h3>
                </div>
                <div
                  className="planets-path"
                >
                  <div
                    className="planets-planet"
                    onClick={[Function]}
                  >
                    <h3>
                      Planets
                    </h3>
                  </div>
                  <div
                    className="people-path"
                  >
                    <div
                      className="people-planet"
                      onClick={[Function]}
                    >
                      <h3>
                        People
                      </h3>
                    </div>
                  </div>
                </div>
              </div>,
            ],
            "className": "favorites-path",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  Favorites
                </h3>,
                "className": "favorites-planet",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": "h3",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="vehicles-planet"
                    onClick={[Function]}
                  >
                    <h3>
                      Vehicles
                    </h3>
                  </div>,
                  <div
                    className="planets-path"
                  >
                    <div
                      className="planets-planet"
                      onClick={[Function]}
                    >
                      <h3>
                        Planets
                      </h3>
                    </div>
                    <div
                      className="people-path"
                    >
                      <div
                        className="people-planet"
                        onClick={[Function]}
                      >
                        <h3>
                          People
                        </h3>
                      </div>
                    </div>
                  </div>,
                ],
                "className": "vehicles-path",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Vehicles
                    </h3>,
                    "className": "vehicles-planet",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Vehicles",
                    },
                    "ref": null,
                    "rendered": "Vehicles",
                    "type": "h3",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="planets-planet"
                        onClick={[Function]}
                      >
                        <h3>
                          Planets
                        </h3>
                      </div>,
                      <div
                        className="people-path"
                      >
                        <div
                          className="people-planet"
                          onClick={[Function]}
                        >
                          <h3>
                            People
                          </h3>
                        </div>
                      </div>,
                    ],
                    "className": "planets-path",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <h3>
                          Planets
                        </h3>,
                        "className": "planets-planet",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Planets",
                        },
                        "ref": null,
                        "rendered": "Planets",
                        "type": "h3",
                      },
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <div
                          className="people-planet"
                          onClick={[Function]}
                        >
                          <h3>
                            People
                          </h3>
                        </div>,
                        "className": "people-path",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <h3>
                            People
                          </h3>,
                          "className": "people-planet",
                          "onClick": [Function],
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "People",
                          },
                          "ref": null,
                          "rendered": "People",
                          "type": "h3",
                        },
                        "type": "div",
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
